{% extends "common/base.html" %}
{% block content %}
<div class="container">
<div class='well'>
<h3>About us</h3>
<p>
PeakWhiz a user-friendly, comprehensive framework that allows users to conduct sophisticated integrative analysis of ChIP-seq data.
Running on a web-based platform, PeakWhiz eliminates the problems associated with platform discrepancies and data management by absorbing the responsibility of data maintenance whilst providing a user-friendly framework for integrative analysis. 
In terms of functionality, PeakWhiz endorses the full range of analyses starting from read mapping to peak calling and subsequent functional analysis. 
By providing support for each stage of processing, data transfer from one analysis to the next is seamless and efficient, requiring little user specified details to derive biologically meaningful results. 
PeakWhiz takes the ease of data transfer to a whole new level with integration with Illumina’s <a href='https://basespace.illumina.com/home/index'>BaseSpace</a> application. BaseSpace is a cloud platform that is directly integrated with Illumina’s sequencing platforms to automatically retrieve results from sequencing runs. 
By removing the hassle of time-consuming manual data transfers, PeakWhiz can obtain the results of these runs and direct them to the ChIP-seq pipeline in an easy and straightforward manner. Alternatively, users can upload their own data sets in a variety of formats to be analyzed ‘on-the- fly’ in a matter of clicks. 
Taken together, PeakWhiz aims to automate and streamline the ChIP-seq analysis pipeline like never before in an attempt to understand the complex mechanisms by which transcription factors and other epigenetic factors work. 
</p>
<br>
<p>
PeakWhiz ChIP-seq Pipeline:
Pre-processing:
1)Read Mapping with Bowtie2
2)Peak Calling with MAC

Integrative Analysis:
1)De Novo Motif Analysis with SEME
2)Motif Enrichment Analysis with CENTDIST
3)Genome Profiling
4)Overlap Analysis with ENCODE ChIP-seq data sets
5)Histone Plot
6)TSS Plot
7)Peak Annotation to Genomic Regions
8)Gene Annotation with GREAT
9)Repeat Analysis
10)Conservation Plot

..and more to come!
</p>
</div>
<div class='media'>
</div>
</div>
{% endblock %}